mindone/utils/misc.py

```diff
@@ -0,0 +1,19 @@
+from inspect import isfunction
+
+import mindspore.ops as ops
+
+
+def extract_into_tensor(a, t, x_shape):
+    b = t.shape[0]
+    out = ops.GatherD()(a, -1, t)
+    return out.reshape(b, *((1,) * (len(x_shape) - 1)))
```

Comment:
Will be good to have docs here, as it's not clear what this function does.

Reply:
Good advice. Clarified in docstring.